src/components/App.jsx

```diff
@@ -1,14 +1,251 @@
-import React from 'react'
+import React, { useState, useEffect, useCallback } from 'react'
+import { useLocation } from 'react-router-dom'
 
+import { BarLeft, BarCenter, BarRight } from 'cozy-bar'
+import { Q, useQuery } from 'cozy-client'
 import { useExternalBridge } from 'cozy-external-bridge/container'
 import flag from 'cozy-flags'
+import { ShareModal } from 'cozy-sharing'
+import BarTitle from 'cozy-ui/transpiled/react/BarTitle'
+import Button from 'cozy-ui/transpiled/react/Buttons'
+import Icon from 'cozy-ui/transpiled/react/Icon'
+import BurgerIcon from 'cozy-ui/transpiled/react/Icons/Burger'
+import PlusIcon from 'cozy-ui/transpiled/react/Icons/Plus'
+import { CircularProgress } from 'cozy-ui/transpiled/react/Progress'
+import Typography from 'cozy-ui/transpiled/react/Typography'
+import useBreakpoints from 'cozy-ui/transpiled/react/providers/Breakpoints'
+
+import styles from '../styles/iframes.styl'
 
 const App = () => {
-  const embeddedDocsUrl = flag('docs.embedded-app-url')
+  // Hooks
+  const { pathname } = useLocation()
+  const { isMobile } = useBreakpoints()
+
+  // Config
+  const driveEnabled = true // Can be replaced by a flag if needed
+
+  // State
+  const [shareModalOpen, setShareModalOpen] = useState(false)
+  const [driveOpen, setDriveOpen] = useState(false)
+  const [controllerHasLoaded, setControllerHasLoaded] = useState(false)
+  const [embeddedAppHasLoaded, setEmbeddedAppHasLoaded] = useState(false)
+
+  // Refs
+  const controllerApp = React.useRef(null)
+  const embeddedApp = React.useRef(null)
+
+  // Data
+  const externalId = pathname.includes('/bridge/docs/')
+    ? pathname.replace('/bridge/docs/', '').replace('/', '')
+    : null
 
+  const { data: files } = useQuery(
+    Q('io.cozy.files')
+      .where({ 'metadata.externalId': externalId })
+      .indexFields(['metadata.externalId'])
+      .limitBy(1),
+    { as: `io.cozy.files/${externalId}` }
+  )
+
+  const currentlyOpenedFile = files && files.length > 0 ? files[0] : null
+
+  // URLs
+  const embeddedDocsUrl = flag('docs.embedded-app-url')
   const { isReady, urlToLoad } = useExternalBridge(embeddedDocsUrl)
 
-  return <iframe id="embeddedApp" src={isReady ? urlToLoad : null}></iframe>
+  const controllerAppUrl = window.location.origin.replace('docs', 'drive')
+
+  // Helpers
+  // Get path without filename and sanitize it
+  const path = currentlyOpenedFile
+    ? currentlyOpenedFile.path.replace(currentlyOpenedFile.name, '')
+    : ''
+  const sanitizedPath = path.endsWith('/') ? path.slice(0, -1) : path
```

Comment:
pretty sure you have all you want in cozy-client. Either is models/file or Filecollection I don't know